kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -551,26 +636,32 @@ An attempt to read more elements than the length of the accessed array is undefi
 
 ```k
   // helper rewrite to implement truncating slices to required size
-  syntax KItem ::= #derefTruncate ( Metadata , ProjectionElems )
+  // `Int`: current stack height
+  syntax KItem ::= #derefTruncate ( Metadata , ProjectionElems , Int )
   // ----------------------------------------------------------------------------------------
+  // current stack height information is useless after truncation
   // no metadata, no change to the value
-  rule <k> #traverseProjection( DEST,         VAL, .ProjectionElems, CTXTS) ~> #derefTruncate(noMetadata, PROJS)
+  rule <k> #traverseProjection( DEST,         VAL, .ProjectionElems, CTXTS) ~> #derefTruncate(noMetadata, PROJS, _)
```

Comment:
This does not quite achieve the goal of remembering the current stack height through the entire traversal.
The job of `#derefTruncate` is to truncate arrays to the length that the metadata indicates.
The projections stored in here are only for dereferencing before that truncation happens. Other `Deref` projections can follow within the traversal after truncating arrays with `#derefTruncate`, and they will need the stack height information (so it is not useless after truncation).